### ISSUE TITLE
when using firebase.analytics() getting .Cannot read property 'getElementsByTagName' of undefined

### ISSUE THREAD

**Issue by vitaly87** [expo/expo#11123]
I am trying to use firebase.analytics,
I installed expo-firebase-analytics and then when runing in on android device getting this errror

Cannot read property 'getElementsByTagName' of undefined

my code is looks like
```

import {firebasConfig} from "./config";
import 'firebase/firestore';
import 'firebase/analytics'

  // Your web app's Firebase configuration
  // For Firebase JS SDK v7.20.0 and later, measurementId is optional
  LogBox.ignoreLogs(['Setting a timer']);

  // Initialize Firebase
  if (firebase.apps.length === 0) {
    firebase.initializeApp(firebasConfig);
    firebase.analytics();


}

```

**Comment by AdamJNavarro** [comment]
Hey @vitaly87, is this only occurring on Android? Also, can you please ensure you've followed the steps for running on the client? https://docs.expo.io/versions/v39.0.0/sdk/firebase-analytics/#expo-client-limitations--configuration

Cheers,
Adam

**Comment by stephenhandley** [comment]
Not sure if this is related but I recently upgraded to 39.0 and I'm getting `Analytics SDK is not available: Firebase JS SDK is not installed: Cannot read property 'length' of undefined` when calling `Analytics.logEvent`

**Comment by ianmartorell** [comment]
@stephenhandley I was getting the same error because I was using firebase 8, while the currently expo supported version (installed with `expo install firebase` is 7.9.0. But I was getting a node-gyp build error while trying to install firebase 7.9, which turns out happened because I was using Node 14. Downgrading to Node 12 allowed me to install firebase 7.9 through `expo install firebase` and the Analytics error went away 😪

**Comment by IjzerenHein** [comment]
The Google `firebase/analytics` package is web-only and doesn't run on native. The correct way to use Firebase Analytics is to use `expo-firebase-analytics`. The `Cannot read property 'length' of undefined` error has already been fixed and support for Firebase 8.0.0 was added recently.
https://github.com/expo/expo/blob/master/packages/expo-firebase-analytics/CHANGELOG.md
